api/src/main/java/org/openmrs/module/patientqueueing/api/dao/PatientQueueingDao.java

```diff
@@ -11,21 +11,84 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.hibernate.Criteria;
+import org.hibernate.criterion.Order;
+import org.hibernate.criterion.Restrictions;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.Provider;
 import org.openmrs.api.db.hibernate.DbSession;
 import org.openmrs.api.db.hibernate.DbSessionFactory;
+import org.openmrs.module.patientqueueing.model.PatientQueue;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Repository;
 
+import java.util.Date;
+import java.util.List;
+
 @Repository("patientqueueing.PatientQueueingDao")
 public class PatientQueueingDao {
 	
-	protected final Log log = LogFactory.getLog(PatientQueueingDao.class);
+	private static final Log log = LogFactory.getLog(PatientQueueingDao.class);
 	
 	@Autowired
 	DbSessionFactory sessionFactory;
 	
 	public DbSession getSession() {
 		return sessionFactory.getCurrentSession();
 	}
+
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService
+	 */
+	public PatientQueue getPatientQueueById(Integer queueId) {
+		return (PatientQueue) getSession().createCriteria(PatientQueue.class)
+		        .add(Restrictions.eq("patientQueueId", queueId)).uniqueResult();
+	}
+
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService
+	 */
+	public List<PatientQueue> getPatientInQueue(Provider provider, Date fromDate, Date toDate, Location location,
+	        Patient patient, String status) {
+		Criteria criteria = getSession().createCriteria(PatientQueue.class);
+		if (fromDate != null && toDate != null) {
+			criteria.add(Restrictions.between("dateCreated", fromDate, toDate));
+		}
+		
+		if (provider != null) {
+			criteria.add(Restrictions.eq("provider", provider));
+		}
+		
+		if (location != null) {
+			criteria.add(Restrictions.eq("locationTo", location));
+		}
+		
+		if (patient != null) {
+			criteria.add(Restrictions.eq("patient", patient));
+		}
+		
+		if (status != null) {
+			criteria.add(Restrictions.eq("status", status));
+		}
+		
+		criteria.addOrder(Order.desc("dateCreated"));
+		
+		return criteria.list();
+	}
+
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService
+	 */
+	public PatientQueue savePatientQueue(PatientQueue patientQueue) {
+		try {
+			sessionFactory.getCurrentSession().saveOrUpdate(patientQueue);
+			return patientQueue;
+		}
+		catch (Exception e) {
```

Comment:
Why don't you just let the exception bubble up to the caller? Not all will test the return value on saving. So they will assume all went well, hence leading to data loss.